wlutil/build.py

```diff
@@ -58,6 +58,38 @@ def handlePostBin(config, linuxBin):
 
        run([config['post-bin'].path] + config['post-bin'].args, env=postbinEnv, cwd=config['workdir'])
 
+def fileDepsTask(name, taskDeps=None, overlay=None, files=None):
+    """Returns a task dict for a calc_dep task that calculates the file
+    dependencies representd by an overlay and/or a list of FileSpec objects.
+    Either can be None.
+    
+    taskDeps should be a list of names of tasks that must run before
+    calculating dependencies (e.g. host-init)"""
+
+    def fileDeps(overlay, files):
+        """The python-action for the filedeps task, returns a dictionary of dependencies"""
+        deps = []
+        if overlay is not None:
+            deps.append(overlay)
+
+        if files is not None:
+            deps += [ f.src for f in files if not f.src.is_symlink() ]
+
+        for dep in deps.copy():
+            if dep.is_dir():
+                deps += [ child for child in dep.glob('**/*') ]
+
+        return { 'file_dep' : [ str(f) for f in deps if not f.is_dir() ] }
+
+    task = {
+            'name' : 'calc_' + name + '_dep',
+            'actions' : [ (fileDeps, [overlay, files]) ],
```

Comment:
If I understand correctly, `fileDeps(overlay, files)` would be called after `img_task_deps` (i.e., `host-init`) and the results dynamically injected into the task graph?  It seems that this resolves #157 without the anticipated limitations.

Reply:
ya, what this is doing is moving the dependency checking to its own task instead of doing it when building the initial dependency graph. That task outputs a list of dynamically generated dependencies (that's what fileDeps does). The nice thing here is that it's a task like any other and I can give it dependencies of its own (host-init in this case).